src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx

```diff
@@ -25,29 +35,73 @@ function AnalyticalEntryInput(props: AnalyticalEntryInputProps) {
         // onChange,
         onRemove,
         index,
+        statementUuid,
+        onAnalyticalEntryDrop,
     } = props;
 
+    const [
+        entryDraggedStatus,
+        setDragStart,
+        setDragEnd,
+    ] = useModalState(false);
+
+    const handleAnalyticalEntryAdd = useCallback(
+        (val: Record<string, unknown> | undefined) => {
+            if (!val) {
+                return;
+            }
+            const typedVal = val as { entryId: number, statementUuid: string };
+            onAnalyticalEntryDrop(typedVal, value.uuid);
+        },
+        [value, onAnalyticalEntryDrop],
+    );
+
+    const dragValue = useMemo(() => ({
+        entryId: value.entry,
+        statementUuid,
+    }), [value.entry, statementUuid]);
+
     // const onFieldChange = useFormObject(index, value, onChange);
 
     return (
-        <div className={styles.entry}>
-            {error?.$internal && (
-                <p>
-                    {error.$internal}
-                </p>
+        <DropContainer
+            className={_cs(
+                styles.dropContainer,
+                entryDraggedStatus && styles.hide,
             )}
-            <h4>
-                {value.entry}
-            </h4>
-            <QuickActionButton
-                name={index}
-                onClick={onRemove}
-                // FIXME: use translation
-                title="Remove Analytical Entry"
+            name="entry"
+            onDrop={!entryDraggedStatus ? handleAnalyticalEntryAdd : undefined}
```

Comment:
Are we disabling drop for the item that is being dragged?
Maybe we should add a comment for this